README.md

```diff
@@ -19,113 +19,122 @@ This library includes the following modules.
 
 # Usage
 
 Cats, ScalaCheck, and Guava type class derivation can be performed both automatically and semi-automatically.
 
-Automatic derivation with implicits:
+Automatic derivation works out of the box by importing implicits from `magnolify.$module.auto._`.
+It works with context bounds i.e. `def plus[T: Semigroup](x: T, y: T)` and implicit paramemters i.e.
+`def f[T](x: T, y: T)(implicit sg: Semigroup[T])`.
+
 ```scala
-case class MyCaseClass(field: NestedCaseClass) 
-case class NestedCaseClass(innerField: String) // works with other non-String elements, this is just an example
+case class Inner(int: Int, str: String)
+case class Outer(inner: Inner)
 
-// implicitly keyword summons an instance of the typeclass wrapping a given case class
-// typeclasses are generated at compile time with macros & Magnolia
-// this works if each element within the case class has a typeclass instance defined within the implicit scope
+// implicitly[T] summons an implicit instance of T
+// - which Magnolia derives at compile time if T is a typeclass F[A]
+// - and A is a case class or ADT
+// - recursively summons typeclasses for case class fields or ADT subtypes
 
-// an example with Cats using the Eq typeclass
+// an example with Cats Semigroup
 import magnolify.cats.auto._
-import cats.kernel.Eq
-// similar to Eq.fromUniversalEquals for a case class but it's at compile time, and don't need == defined reasonably
-// instead of using == in fromUniversalEquals, you need an Eq instance for each part of your case class
-implicit val eqString: Eq[String] = ??? // could be fromUniversalEquals, but doesn't have to be
-val eqCaseClass: Eq[MyCaseClass] = implicitly[Eq[MyCaseClass]] 
-
-// an example with ScalaCheck using the Arbitrary typeclass
-// this example is a bit more practical, since there's no typeclass instantiation built into Scalacheck, unlike Cats
+import cats._
+import cats.instances.all._ // implicit instances for Semigroup[Int], etc.
+val sg: Semigroup[Outer] = implicitly[Semigroup[Outer]]
+sg.combine(Outer(Inner(1, "hello, ")), Outer(Inner(100, "world!")))
+// = Outer(Inner(101,hello, world!))
+
+// an example with ScalaCheck Arbitrary
 import magnolify.scalacheck.auto._
-import org.scalacheck.Arbitrary
-implicit val arbString: Arbitrary[String] = ??? // could be Arbitrary.arbitrary[String]
-val arbitraryCaseClass: Arbitrary[MyCaseClass] = implicitly[Arbitrary[MyCaseClass]]
-
-// an example with Guava using the Funnel typeclass
-import magnolify.guava.auto._
-import com.google.common.hash.Funnel
-implicit val funnelString: Funnel[String] = ??? // could be Funnels.stringFunnel(charset)
-val implicitly[Funnel[MyCaseClass]]
+import org.scalacheck._ // implicit instances for Arbitrary[Int], etc.
+val arb: Arbitrary[Outer] = implicitly[Arbitrary[Outer]]
+arb.arbitrary.sample
+// = Some(Outer(Inter(12345, abcde)))
+
+// an example with Guava Funnel
+import magnolify.guava.auto._ // includes implicit instances for Funnel[Int], etc.
```

Comment:
Importing `cats.instances.all._`, `org.scalacheck._` & `magnolify.guava.auto._` should import all primitive instances and is the more idiomatic.